CoreAardvark/ARKExceptionLogging.m

```diff
@@ -31,24 +32,33 @@
 
 NSMutableArray *_Nullable ARKUncaughtExceptionLogDistributors = nil;
 
+// Use aliases for getting/setting the uncaught exception handler so we can replace these in unit tests.
+ARKUncaughtExceptionHandlerGetter ARKGetUncaughtExceptionHandler = NSGetUncaughtExceptionHandler;
+ARKUncaughtExceptionHandlerSetter ARKSetUncaughtExceptionHandler = NSSetUncaughtExceptionHandler;
+
 NSLock * ARKGetUncaughtExceptionLogDistributorsLock()
 {
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        ARKUncaughtExceptionLogDistributorsLock = [[NSLock alloc] init];
+        ARKUncaughtExceptionLogDistributorsLock = [NSLock new];
     });
     return ARKUncaughtExceptionLogDistributorsLock;
 }
 
 void ARKHandleUncaughtException(NSException *exception)
 {
+    NSLock *const logDistributorsLock = ARKGetUncaughtExceptionLogDistributorsLock();
+    [logDistributorsLock lock];
+
     for (ARKLogDistributor *const logDistributor in ARKUncaughtExceptionLogDistributors) {
         [logDistributor logWithType:ARKLogTypeError userInfo:nil format:@"Uncaught exception '%@':\n%@", exception.name, exception.debugDescription];
         [logDistributor distributeAllPendingLogsWithCompletionHandler:^{}];
         [logDistributor waitUntilAllPendingLogsHaveBeenDistributed];
     }
-    
-    if (ARKPreviousUncaughtExceptionHandler != nil) {
+
+    [logDistributorsLock unlock];
```

Comment:
`unlock` should go after we access `ARKPreviousUncaughtExceptionHandler` below

Reply:
Moved the reading of `ARKPreviousUncaughtExceptionHandler` inside the lock, but left the calling outside, to avoid deadlock in case the previous handler somehow calls back into this one.